src/libraries/System.IO.FileSystem/tests/FileStream/ctor_options_as.cs

```diff
@@ -27,6 +27,18 @@ protected override FileStream CreateFileStream(string path, FileMode mode, FileA
                         PreallocationSize = PreallocationSize
                     });
 
+        protected override FileStream CreateFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options)
+            => new FileStream(path,
+                    new FileStreamOptions
+                    {
+                        Mode = mode,
+                        Access = access,
+                        Share = share,
+                        BufferSize = bufferSize,
+                        Options = options,
+                        PreallocationSize = PreallocationSize
+                    });
+
         protected FileStreamOptions GetOptions(FileMode mode, FileAccess access, FileShare share, FileOptions options, long preAllocationSize)
```

Comment:
```suggestion
        protected FileStreamOptions GetOptions(FileMode mode, FileAccess access, FileShare share, FileOptions options, int bufferSize = 4096, long preAllocationSize = 0)
```

Reply:
I would need to update all usages of `GetOptions` where `preallocationSize` is currently passed as the first argument, otherwise this change would introduce a silent error (instead of passing numbers as preallocation size, we would be passing them as buffer size:

```cs
GetOptions(mode, FileAccess.Write, FileShare.None, FileOptions.None, 123) // preallocationSize = 123 would became bufferSize = 123 and preallocationSize = 0
```

Reply:
What about this?
```suggestion
        protected FileStreamOptions GetOptions(FileMode mode, FileAccess access, FileShare share, FileOptions options, long preAllocationSize = 0, int bufferSize = 4096)
```